jams/core.py

```diff
@@ -693,8 +693,8 @@ def append(self, time=None, duration=None, value=None, confidence=None):
         >>> ann.append(time=3, duration=2, value='E#')
         '''
 
-        self.data.add(Observation(time=time,
-                                  duration=duration,
+        self.data.add(Observation(time=float(time),
+                                  duration=float(duration),
                                   value=value,
                                   confidence=confidence))
```

Comment:
Yup.  The reasoning here is confidence is namespace-dependent; sometimes it's required (tempo), sometimes it's optional (msd_tagtraum).